osclib/core.py

```diff
@@ -414,26 +414,43 @@ def attribute_value_save(apiurl, project, name, value, namespace='OSRT', package
     url = makeurl(apiurl, list(filter(None, ['source', project, package, '_attribute'])))
     http_POST(url, data=ET.tostring(root))
 
+
 def attribute_value_delete(apiurl, project, name, namespace='OSRT', package=None):
     http_DELETE(makeurl(
         apiurl, list(filter(None, ['source', project, package, '_attribute', namespace + ':' + name]))))
 
-@memoize(session=True)
+
 def repository_path_expand(apiurl, project, repo):
     """Recursively list underlying projects."""
+    visited_repos = set(f"{project}:{repo}")
```

Comment:
That's already done by `_recursive_repository_path_expand`, FWICT the split into separate functions isn't actually needed?

Reply:
the split isn't needed, I can do it with a default argument as well. changed.